File: algorithm/typescript/BOJ1764.ts
// problem : https://www.acmicpc.net/problem/1764

import fs = require("fs");

const input: string[] = fs.readFileSync("/dev/stdin").toString().split("\n");
const NM: number[] = input[0].split(" ").map(Number);
const N: number = NM[0];
const M: number = NM[1];
const result: string[] = [];
const map = new Map();

for (let i = 1; i <= N; i++) {
  map.set(input[i], 1);
}

for (let i = N + 1; i <= N + M; i++) {
  if (map.has(input[i])) {
    result.push(input[i]);
  }
}

result.sort();
console.log(result.length);
console.log(result.join("\n"));

File: algorithm/python/BOJ1764.py
# problem : https://www.acmicpc.net/problem/1764

import sys

N, M = map(int, sys.stdin.readline().split())
name_dict = dict()
result = list()

for i in range(0, N):
    name = sys.stdin.readline().rstrip()
    name_dict[name] = 1

for i in range(0, M):
    name = sys.stdin.readline().rstrip()
    if name_dict.get(name) != None:
        result.append(name)

result.sort()
print(len(result))
for name in result:
    print(name)
